nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ExtractGrok.java

```diff
@@ -234,6 +247,11 @@ public void onStopped() {
 
     @OnScheduled
     public void onScheduled(final ProcessContext context) throws GrokException, IOException {
+
+        if (context.getProperty(KEEP_EMPTY_CAPTURES).isSet()) {
```

Comment:
with required=true, you would not need to check if the property is set anymore

Reply:
done